File: Backend/src/main/java/com/company/Billing/entity/Invoice.java
package com.company.Billing.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "invoices")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate invoiceDate;

    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal totalSubtotal;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal totalTaxes;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal totalInvoice;

    @OneToMany(mappedBy = "invoice", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<InvoiceItem> items = new ArrayList<>();
}

File: Frontend/src/app/models/invoice.ts
import { Customer } from './customer';
import { Product } from './product';

export interface Invoice {
  id?: number;
  invoiceDate: string;
  customer: Customer;
  items: InvoiceItem[];
  totalSubtotal: number;
  totalTaxes: number;
  totalInvoice: number;
}

export interface InvoiceItem {
  id?: number;
  product: Product;
  quantity: number;
  unitPrice: number;
  taxRate: number;
  subtotal: number;
  taxAmount: number;
  totalLine: number;
}
